### PR TITLE
restructure storage samples

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/samples/CMakeLists.txt
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/samples/CMakeLists.txt:16]:
```suggestion
target_link_libraries(transactional_checksum PRIVATE azure-storage-blobs)

```

File: sdk/storage/azure-storage-common/CMakeLists.txt
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-common/CMakeLists.txt:131]:
```suggestion
endif()

```

File: sdk/storage/azure-storage-files-datalake/samples/CMakeLists.txt
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/samples/CMakeLists.txt:7]:
```suggestion
target_link_libraries(datalake_getting_started PRIVATE azure-storage-files-datalake)

```

File: sdk/storage/azure-storage-blobs/samples/CMakeLists.txt
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/samples/CMakeLists.txt:16]:
```suggestion
add_executable(blob-getting-started blob_getting_started.cpp)
target_link_libraries(blob-getting-started PRIVATE azure-storage-blobs)

add_executable(blob-list-operation blob_list_operation.cpp)
target_link_libraries(blob-list-operation  PRIVATE azure-storage-blobs)

add_executable(blob-sas blob_sas.cpp)
target_link_libraries(blob-sas PRIVATE azure-storage-blobs)

add_executable(transactional-checksum transactional_checksum.cpp)
target_link_libraries(transactional-checksum PRIVATE azure-storage-blobs)
```

We use `name-dash-name` for CMake targets ( libs and exe )

File: sdk/storage/azure-storage-blobs/samples/CMakeLists.txt
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/samples/CMakeLists.txt:4]: nit. Ideally, we might want to have one folder for each sample. Each folder would contain a CMakeList and the source code.

In the future, we want to make then entire folder a copy/paste project with no dependency to the repo.

This works for now and is better than previous approach.

File: sdk/storage/azure-storage-blobs/samples/blob_getting_started.cpp
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/samples/blob_getting_started.cpp:18]: please keep the logic for reading the connection string from env-var.  We need it to be able to run the samples during CI gates.
And it is also ideal and consistent for customers ( avoid recompile for switching accounts )

File: sdk/storage/azure-storage-blobs/samples/blob_getting_started.cpp
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/samples/blob_getting_started.cpp:42]: nit. Instead of letting app go to unhandled exception, add try/catch to show catching Storage exception.

File: sdk/storage/azure-storage-blobs/samples/blob_sas.cpp
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/samples/blob_sas.cpp:13]: Read this from env-vars as well.  The current `test-resources` defines  `ACCOUNT_NAME` and `ACCOUNT_KEY`.  We should consider updating this to `AZURE_STORAGE_ACCOUNT_NAME` and `AZURE_STORAGE_ACCONT_KEY`.

File: sdk/storage/azure-storage-files-datalake/samples/CMakeLists.txt
Comment (vhvb1989) [inline sdk/storage/azure-storage-files-datalake/samples/CMakeLists.txt:7]:
```suggestion
add_executable(datalake-getting-started datalake_getting_started.cpp)
target_link_libraries(datalake-getting-started PRIVATE azure-storage-files-datalake)
```

File: sdk/storage/azure-storage-files-shares/samples/CMakeLists.txt
Comment (vhvb1989) [inline sdk/storage/azure-storage-files-shares/samples/CMakeLists.txt:7]:
```suggestion
add_executable(file-share-getting-started file_share_getting_started.cpp)
target_link_libraries(file-share-getting-started PRIVATE azure-storage-files-shares)
```

File: sdk/storage/azure-storage-queues/samples/CMakeLists.txt
Comment (vhvb1989) [inline sdk/storage/azure-storage-queues/samples/CMakeLists.txt:10]: update to target-name style

**Review by vhvb1989 — COMMENTED** [review]
Make samples to read from environment for connection string or account name/key
Use the same env-vars defined in the `test-resources.json` (update the name in the test-resources.json if required) so we can run samples during nightly CI gates

File: sdk/storage/azure-storage-queues/samples/CMakeLists.txt
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-queues/samples/CMakeLists.txt:7]:
```suggestion
add_executable(queue-getting-started queue_getting_started.cpp)
target_link_libraries(queue-getting-started PRIVATE azure-storage-queues)
```

File: sdk/storage/azure-storage-queues/samples/CMakeLists.txt
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-queues/samples/CMakeLists.txt:10]:
```suggestion
add_executable(queue-encode-message queue_encode_message.cpp)
target_link_libraries(queue-encode-message PRIVATE azure-storage-queues)
```

**Review by antkmsft — APPROVED** [review]

File: sdk/storage/azure-storage-blobs/samples/blob_sas.cpp
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/samples/blob_sas.cpp]: I'll just read from `AZURE_STORAGE_CONNECTION_STRING` and parse account name and key by myself.

File: sdk/storage/azure-storage-blobs/samples/blob_getting_started.cpp
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/samples/blob_getting_started.cpp:63]: I don't think it's a good habit to enclose many API calls with one try/catch block. It would be hard to tell which exactly API failed even if there was an exception, and the user wouldn't be able to figure out which APIs succeeded, and where to resume.

If we had one try/catch block for each API call, that would make the sample too lengthy, and our customers wouldn't be able to focus on the code we want to demonstrate.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/samples/blob_getting_started.cpp:63]: I'm thinking, maybe we should have an separate sample for how to catch storage exceptions.

**Comment by Jinming-Hu** [comment]
@vhvb1989 I just added support for reading from env var. It all looks good to me now. Feel free to hit the merge button to unblock youself.

File: sdk/storage/azure-storage-blobs/samples/blob_getting_started.cpp
Comment (vhvb1989) [reply sdk/storage/azure-storage-blobs/samples/blob_getting_started.cpp:63]: > It would be hard to tell which exactly API failed even if there was an exception

I would expect customers would turn logging to console on while trying to debug issues.  Then, the logging policy would tell what's the I/O request before throwing the exception.

But, yes,  this can be done later. Not a blocker for this PR

**Pull request merged**
